project-minimization-core/src/main/kotlin/org/plan/research/minimization/core/model/DDInfo.kt

```diff
@@ -0,0 +1,28 @@
+package org.plan.research.minimization.core.model
+
+import java.util.IdentityHashMap
+
+/**
+ * Interface representing additional information for DD algorithm.
+ */
+interface DDInfo<T : DDItem> {
+    fun of(item: T): DDItemInfo
+
+    fun importanceOf(items: List<T>): IdentityHashMap<T, Boolean> {
+        val result = IdentityHashMap<T, Boolean>()
+        items.forEach { result[it] = of(it).likelyImportant }
+        return result
+    }
+
+    companion object {
+        fun <T : DDItem> empty(): DDInfo<T> = object : DDInfo<T> {
+            override fun of(item: T): DDItemInfo = DDItemInfo()
+        }
```

Comment:
It's better to make something like `object EmptyDDInfo : DDInfo`

Reply:
The problem is that it is a polymorphic value. So in order to make a singleton of empty data I have to make the interface covariant in T which in turn prohibits importanceOf function in the interface, so it has to be made an extension.

I did that, but IMHO it complicates code. Merge if its okay for you